java/lance-namespace-glue/src/main/java/com/lancedb/lance/namespace/glue/LanceGlueNamespace.java

```diff
@@ -0,0 +1,248 @@
+import com.lancedb.lance.namespace.model.NamespaceExistsRequest;
+import com.lancedb.lance.namespace.model.NamespaceExistsResponse;
+import com.lancedb.lance.namespace.model.QueryRequest;
+import com.lancedb.lance.namespace.model.RegisterTableRequest;
+import com.lancedb.lance.namespace.model.RegisterTableResponse;
+import com.lancedb.lance.namespace.model.TableExistsRequest;
+import com.lancedb.lance.namespace.model.TableExistsResponse;
+import com.lancedb.lance.namespace.model.UpdateTableRequest;
+import com.lancedb.lance.namespace.model.UpdateTableResponse;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Sets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.glue.GlueClient;
+import software.amazon.awssdk.services.glue.model.GetDatabasesRequest;
+import software.amazon.awssdk.services.glue.model.GetDatabasesResponse;
+
+import java.io.Closeable;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class LanceGlueNamespace implements LanceNamespace, Closeable {
+  private static final Logger LOG = LoggerFactory.getLogger(LanceGlueNamespace.class);
+  private static final int MAX_GLUE_LISTING_SIZE = 100;
+
+  private GlueProperties properties;
+  public GlueClient glueClient;
+  private String name;
+
+  public LanceGlueNamespace() {}
+
+  @Override
+  public void initialize(String name, Map<String, String> properties) {
+    GlueProperties glueProperties = new GlueProperties(properties);
+    GlueClient glueClient =
+        GlueClient.builder().applyMutation(glueProperties::configureClientBuilder).build();
+    initialize(name, glueProperties, glueClient);
+  }
+
+  @VisibleForTesting
+  void initialize(String name, GlueProperties properties, GlueClient glueClient) {
+    this.properties = properties;
+    this.glueClient = glueClient;
+    this.name = name;
+  }
+
+  @Override
+  public ListNamespacesResponse listNamespaces(ListNamespacesRequest request) {
+    validateSingleLevelNamespace(request.getParent());
+
+    GetDatabasesRequest.Builder listRequest =
+        GetDatabasesRequest.builder().catalogId(properties.glueCatalogId());
+    int pageSize = request.getPageSize() != null ? request.getPageSize() : Integer.MAX_VALUE;
+    int remaining = pageSize;
+    String glueNextToken = request.getPageToken();
+    Set<String> databases = Sets.newHashSet();
+    do {
+      int fetchSize = Math.min(remaining, MAX_GLUE_LISTING_SIZE);
+      GetDatabasesResponse response =
+          glueClient.getDatabases(
+              listRequest.maxResults(fetchSize).nextToken(glueNextToken).build());
+      response.databaseList().forEach(d -> databases.add(d.name()));
+      glueNextToken = response.nextToken();
+      remaining = pageSize - databases.size();
+    } while (glueNextToken != null && remaining > 0);
+
+    LOG.debug("Listing namespace returned : {}", String.join(", ", databases));
```

Comment:
this debug is just returning the response, I think we don't need to log that information.